rules/Renaming/Rector/Namespace_/RenameNamespaceRector.php

```diff
@@ -124,6 +163,75 @@ public function configure(array $configuration): void
         $this->oldToNewNamespaces = $configuration;
     }
 
+    private function processChangeDocblock(Node $node): ?Node
+    {
+        $phpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($node);
+        $phpDocNode = $phpDocInfo->getPhpDocNode();
+        $children = $phpDocNode->children;
+
+        foreach ($children as $child) {
+            if (! $child instanceof PhpDocTagNode) {
+                continue;
+            }
+
+            $value = $child->value;
+            if (! in_array($value::class, self::TO_BE_CHANGED, true)) {
+                continue;
+            }
+
+            /** @var ReturnTagValueNode|VariadicAwareParamTagValueNode|VarTagValueNode $value */
+            $this->refactorDocblock($value);
+        }
+
+        if (! $phpDocInfo->hasChanged()) {
+            return null;
+        }
+
+        return $node;
+    }
+
+    private function refactorDocblock(
+        ReturnTagValueNode|VariadicAwareParamTagValueNode|VarTagValueNode $value
+    ): void {
+        /** @var ReturnTagValueNode|VariadicAwareParamTagValueNode|VarTagValueNode $value */
+        $types = $value->type instanceof BracketsAwareUnionTypeNode
+            ? $value->type->types
+            : [$value->type];
+
+        foreach ($types as $key => $type) {
+            if (! $type instanceof IdentifierTypeNode) {
+                continue;
+            }
+
+            $name = $type->name;
+            $trimmedName = ltrim($type->name, '\\');
+
+            if ($name === $trimmedName) {
+                continue;
+            }
+
+            $renamedNamespaceValueObject = $this->namespaceMatcher->matchRenamedNamespace(
+                $trimmedName,
+                $this->oldToNewNamespaces
+            );
+            if (! $renamedNamespaceValueObject instanceof RenamedNamespace) {
+                continue;
+            }
+
+            $newType = new IdentifierTypeNode('\\' . $renamedNamespaceValueObject->getNameInNewNamespace());
+
+            if ($value->type instanceof BracketsAwareUnionTypeNode) {
+                $types[$key] = $newType;
+            } else {
+                $value->type = $newType;
+            }
+        }
+
+        if ($value->type instanceof BracketsAwareUnionTypeNode) {
+            $value->type->types = $types;
+        }
+    }
+
```

Comment:
Can you extract this logic to external service? The rule is  getting too complex

Reply:
implemented, I created `DocBlockNamespaceRenamer` service for it https://github.com/rectorphp/rector-src/pull/1917/commits/17838a83b04ca1eb2f947dc8a45ea957cefd4c68